.github/workflows/docker.yaml

```diff
@@ -0,0 +1,104 @@
+name: Docker
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: hadolint/hadolint-action@v3.1.0
+        with:
+          recursive: true
+          ignore: DL3018
+
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Workaround: https://github.com/docker/build-push-action/issues/461
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@v3.0.0
+
+    #   - name: Log in to Docker Hub
+    #     if: github.event_name != 'pull_request'
+    #     uses: docker/login-action@v3.0.0
+    #     with:
+    #       username: ${{ secrets.DOCKERHUB_USERNAME }}
+    #       password: ${{ secrets.DOCKERHUB_TOKEN }}
```

Comment:
I now set these two as repository secrets. So theoretically this should work.

Reply:
I am not sure how to enable this file here and see it running already as part of the PR...

Reply:
I will enable this in next patch, no problem, once you added secrets, should be easy for me to add this

Reply:
opened #876